src/main/java/org/xbill/DNS/NioUdpClient.java

```diff
@@ -169,6 +229,9 @@ private void silentCloseChannel() {
 
               addr = new InetSocketAddress(local.getAddress(), port);
             }
+            if (addr.getPort() == SimpleResolver.RESERVED_MDNS_PORT) {
```

Comment:
I kind of disagree with the statement that "port 5353 is valid in normal DNS". Since mDNS is becoming more prevalent, it would be recommended not to use that port accidentally to avoid receiving requests from other hosts when you're expecting replies. And it doesn't _hurt_ to avoid it, since this part of the code no longer differentiates between receiving DNS and mDNS replies.

Reply:
In regular DNS the socket is bound, so the only replies accepted are from a DNS server. The case shouldn't occur when using defaults though.

But `addr==null` needs to be handled anyway.